CHANGELOG.md

```diff
@@ -1,3 +1,30 @@
+* Allow for providing a reason when aborting in `renderToPipeableStream` or `renderToReadableStream`. The reason will be passed to `onError` on the server. If a reason is not provided a default error messages is provided to `onError` instead ([@gnoff](https://github.com/gnoff)) in [#24680](https://github.com/facebook/react/pull/24680)
+* Eliminate extraneous text separators whenever possible when Server Side Rendering with `renderToPipeableStream` or `renderToReadableStream` ([@gnoff](https://github.com/gnoff)) in [#24630](https://github.com/facebook/react/pull/24630)
+* Disallow complex children inside `<title>` elements to better align with browser constraints requiring only `Text` content within this element type. ([@gnoff](https://github.com/gnoff)) in [#24679](https://github.com/facebook/react/pull/24679)
+* Fix default `highWaterMark` behavior of `ReadableStream` in `renderToReadableStream` ([@jplhomer](https://github.com/jplhomer)) in [#24641](https://github.com/facebook/react/pull/24641)
+
+### Server Components (Experimental)
+
+* **useId Hook** `useId` is now supported in Server Components. useId's uniqueness guarantees are different in Server Components because there is no client hydration but there can be ID collisions across refetches if you refetch only a portion of the application on subsequent requests. To avoid collisions `useId` will use an option `identifierPrefix` which needs to be chose to ensure uniqueness across multiple requests. An additional nuanced behavior is that if you pass a Server Component using `useId` to a Client Component and that client component renders the passed in element into more than one place, the id's within the Server Component elements will not be unique. In the future we may add a warning in Dev if you render a Server Component that uses `useId` into more than one place on the client. ([@gnoff](https://github.com/gnoff)) in [#24172](https://github.com/facebook/react/pull/24172)
```

Comment:
Since Server Components are experimental it's enough for this description to live in PR. Whereas the changelog entry could be just like

- Add support for `useId()` inside Server Components.